trio/_core/_tests/test_asyncgen.py

```diff
@@ -48,7 +52,7 @@ async def async_main():
         assert collected.pop() == "abandoned"
 
         # aclosing() ensures it's cleaned up at point of use
-        async with contextlib.aclosing(example("exhausted 1")) as aiter:
+        async with contextlib.aclosing(example("exhausted 1")) as aiter:  # type: ignore[attr-defined]  # skipif makes sure we are at least 3.10
```

Comment:
It's possible to use https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks
though unfortunately you can't just plop an `assert sys.version_info > 3.X` in the function body (is it possible to do `if sys.version_info > 3.x; return`?). If it requires indenting the whole test with a redundant check it's probably not worth it.
The code then also won't be type-checked in CI at all since we're specifying the version, so maybe net loss overall.

Reply:
Changed in 9c09d80

Reply:
It might be easier to just avoid using `aclosing()`, it's a trivial function to inline:
```py
aiter = example("exhausted 1")
try:
    assert 42 == await aiter.asend(None)
finally:
    await aiter.close()
```
Or import it from `async_generator`.

Reply:
Inlined in 5da3819